packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -34,22 +42,38 @@ function isValidPair<K extends keyof SearchParameters>(
 }
 
 function isValidKey(key: string): key is keyof SearchParameters {
-  const validKeys: (keyof SearchParameters)[] = ['q', 'enableQuerySyntax'];
-  return validKeys.some((validKey) => validKey === key);
+  const supportedParameters: Record<
+    keyof Required<SearchParameters>,
+    boolean
+  > = {
+    q: true,
+    aq: true,
+    cq: true,
+    enableQuerySyntax: true,
+    firstResult: true,
+    numberOfResults: true,
+    sortCriteria: true,
+  };
```

Comment:
Why use a `Record`? Is there a situation where the `boolean` value of a key could be false? If not, I think the array used before is better?

Reply:
I agree, however I switched to use an object so that typescript can warn us to update this object when a new parameter is supported. I was unable to find a way to do this when using an array. The `boolean` value can be either `true` or `false`; it is not important for the logic.